src/rust-crypto/libolm_migration.ts

```diff
@@ -175,9 +175,20 @@ async function migrateBaseData(
                 await sleep(2000);
             }
         }
-        if (backupInfo) {
-            migrationData.backupVersion = backupInfo.version;
-            migrationData.backupRecoveryKey = recoveryKey;
+        if (backupInfo && backupInfo.algorithm == "m.megolm_backup.v1.curve25519-aes-sha2") {
```

Comment:
I think you should update the comment above "All we can really do is ask the server for the most recent version." since we are now doing more?